backend/src/tasks/LookingGlass.ts

```diff
@@ -0,0 +1,286 @@
+  const Vulnerabilities: Vulnerability[] = [];
+
+  for (const line of collections) {
+    //Match the organization to the LookingGlass Collection
+    if (organizationName == line['name']) {
+      const collectionID = line['id'];
+      console.log(line['name']);
+      //Query LookingGlass for the Threat info
+      const data: LGThreatResponse[] = await getThreatInfo(collectionID);
+      const responseDomains: Domain[] = await saveAndPullDomains(
+        data,
+        organizationId,
+        scanId,
+        Org
+      );
+      for (const l of data['results']) {
+        //Create a dictionary of relevant fields from the API request
+        if (typeof Org === 'object') {
+          const val = {
+            firstSeen: new Date(l['firstSeen']),
+            lastSeen: new Date(l['lastSeen']),
+            sources: l['sources'],
+            ref_type: l['ref']['type'],
+            ref_right_type: l['ref']['right']['type'],
+            ref_right_id: l['ref']['right']['id'],
+            ref_left_type: l['ref']['left']['type'],
+            ref_left_id: l['ref']['right']['id'],
+
+            right_ticScore: l['right']['ticScore'],
+            right_classifications: l['right']['classifications'],
+            right_name: l['right']['name'],
+
+            left_type: l['left']['type'],
+            left_ticScore: l['left']['ticScore'],
+            left_name: l['left']['name']
+          };
+          if (l['right']['classifications'][0] == 'Vulnerable Service') {
+            val['vulnOrMal'] = 'Vulnerability';
+          } else {
+            val['vulnOrMal'] = 'Malware';
+          }
+
+          //if the Domain already exists and add this vuln to the associated vulnerability
+          if (ipsAndDomains.includes(l['left']['name'])) {
+            for (const Vuln of Vulns_list) {
+              if (l['left']['name'] == Vuln['domain'].name) {
+                Vuln['structuredData']['lookingGlassData'].push(val);
+              }
+            }
+          }
+          //if the Domain hasn't been used create a new Domain and a new Vulnerability
+          else {
+            for (const x of responseDomains) {
+              if (x.name == l['left']['name']) {
+                const current_Domain = x;
+                const V = {
+                  domain: current_Domain,
+                  lastSeen: new Date(Date.now()),
+                  title: 'Looking Glass Data',
+                  state: 'open',
+                  source: 'lookingGlass',
+                  needsPopulation: false,
+                  structuredData: {
+                    lookingGlassData: [val]
+                  },
+                  description: `These are Vulnerabilities and Malware found by LookingGlass for ${organizationName}`
+                };
+                Vulns_list.push(V);
+              }
+            }
+            ipsAndDomains.push(l['left']['name']);
+          }
+        }
+      }
+      for (const v of Vulns_list) {
```

Comment:
This feels redundant. Instead of maintaining two lists `Vulns_list` and `Vulnerabilities`, let's just create one list called `vulnerabilities`, and we can replace `Vulns_list.push(V);` with `vulnerabilities.push(plainToClass(Vulnerability, v));`

Reply:
Just wanted to verify this will work, can i still add new values to the structured data, if it is of type Vulnerability, right now I have been waiting to convert it until after I have added all the vulns associated with the IP/Domain, we can jump on a call to talk about it if it doesn't make sense

Reply:
Yeah, we can still add new values to the structured data even if it's of type Vulnerability.